### PR TITLE
Add a bunch of docs

### PR DESCRIPTION
**Pull request by phofl**

### CODE REVIEW

File: dask_expr/io/parquet.py [hunk 1]
```diff
@@ -330,6 +330,115 @@ def to_parquet(
...
+        If False (default), construct data-set from scratch. If True, add new
+        row-group(s) to an existing data-set. In the latter case, the data-set
+        must exist, and the schema must match the input data.
+    overwrite : bool, default False
+        Whether or not to remove the contents of `path` before writing the dataset.
+        The default is False.  If True, the specified path must correspond to
+        a directory (but not the current working directory).  This option cannot
+        be set to True if `append=True`.
+        NOTE: `overwrite=True` will remove the original data even if the current
+        write operation fails.  Use at your own risk.
```
Comment (hendrikmakait) [inline dask_expr/io/parquet.py:361]: Out of scope: At first glance, these two options appear to be the inverse of one another. That is not the case though. We may want to do a better job at explaining this. (I still don't understand it.)

**Review by hendrikmakait — APPROVED** [review]
Thanks, @phofl!

File: dask_expr/io/parquet.py (hunk 1 above)
Comment (phofl) [reply dask_expr/io/parquet.py:361]: Yeah the parquet docs are ridiculously bad at actually explaining at what every option does

Comment (hendrikmakait) [reply dask_expr/io/parquet.py:361]: I addded https://github.com/dask/dask-expr/issues/1054 as a reminder.

**Pull request merged**